dht/table.go

```diff
@@ -227,6 +241,40 @@ func (table *InMemTable) NumPeers() int {
 	return len(table.addrsBySignatory)
 }
 
+func (table *InMemTable) HandleExpired(peerID id.Signatory) bool {
+	table.expiryBySignatoryMu.Lock()
+	defer table.expiryBySignatoryMu.Unlock()
+	expiry, ok := table.expiryBySignatory[peerID]
+	if !ok {
+		return false
+	}
+	expired := (time.Now().Sub(expiry.timestamp)) > expiry.minimumExpiryAge
+	if expired {
+		table.DeletePeer(peerID)
+		table.DeleteExpiry(peerID)
```

Comment:
This causes a deadlock, because `DeleteExpiry` will attempt to call `table.expiryBySignatoryMu.Lock()` which is already being held by `HandleExpiry`. Tests are needed, because this would've been caught straight away.

Reply:
Adding tests in a bit